serverless/knative_eventing/serverless-kn-trigger.adoc

```diff
@@ -0,0 +1,74 @@
+include::modules/common-attributes.adoc[]
+include::modules/serverless-document-attributes.adoc[]
+:context: serverless-kn-trigger
+
+toc::[]
+
+[id="serverless-kn-trigger"]
+= Using triggers with Knative Eventing
+
+All events which are sent to a channel or broker will be sent to all subscribers of that channel or broker by default.
+
+Using triggers allows you to filter events from a channel or broker, so that subscribers will only receive a subset of events based on your defined criteria.
+
+The Knative client provides a set of `kn trigger` commands that can be used to create and manage triggers.
+
+.Prerequisites
+Before you can use triggers, you will need:
+
+* Knative Eventing and `kn` installed.
+* An available broker (either the `default` broker, or one that you have created).
+* An available event consumer, for example, a Knative service.
+
+.Example trigger YAML:
+
+----
+apiVersion: eventing.knative.dev/v1alpha1
+kind: Trigger
+metadata:
+  name: trigger-example <1>
+spec:
+ broker: default <2>
+ subscriber:
+    ref:
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: my-service <3>
+----
+
+<1> The name of the trigger being created.
+<2> The name of the broker where events will be filtered from. If the broker is not specified, the trigger will revert to using the `default` Broker.
+<3> The name of the service that will consumer filtered events.
+
+== Creating a trigger using `kn`
+
+.Procedure
+
+Create a trigger:
+----
+kn trigger create <TRIGGER-NAME> --broker <BROKER-NAME> --filter <KEY=VALUE> --sink <SINK>
+----
+
+include::modules/kn-trigger-list.adoc[leveloffset=+1]
+include::modules/kn-trigger-describe.adoc[leveloffset=+1]
+
+== Deleting a trigger using `kn`
+
+.Procedure
+
+Delete a trigger:
+----
+kn trigger delete <TRIGGER-NAME>
+----
+
+== Updating a trigger using `kn`
+
+.Procedure
+
+Update a trigger:
+----
+kn trigger update <TRIGGER-NAME>
+----
+
+// what does this do exactly? What does it update? Steps required before?
```

Comment:
@rhuss @navidshaikh can you please advise on how the update command works? Presumably a user would update a yaml file or something first and then run this command? Thanks!

Reply:
A user can update sink service of the trigger or add/remove/update filters of the trigger.
```
✗ kn trigger update -h                                                      
Update a trigger

Usage:
  kn trigger update NAME --filter KEY=VALUE --sink SINK [flags]

Examples:

  # Update the filter which key is 'type' to value 'knative.dev.bar' in a trigger 'mytrigger'
  kn trigger update mytrigger --filter type=knative.dev.bar

  # Remove the filter which key is 'type' from a trigger 'mytrigger' 
  kn trigger update mytrigger --filter type-

  # Update the sink of a trigger 'mytrigger' to 'svc:new-service'
  kn trigger update mytrigger --sink svc:new-service
```

Reply:
Awesome, thank you so much!

Reply:
No YAML file update required as such.